src/neoxp/Node/NodeUtility.cs

```diff
@@ -138,5 +142,127 @@ public static void SignOracleResponseTransaction(ProtocolSettings settings, Expr
 
             return tx;
         }
+
+        public const byte Prefix_Contract = 8;
+        private const byte Prefix_NextAvailableId = 15;
+
+        private static int LastUsedContractId(DataCache snapshot)
+        {
+            StorageKey key = new KeyBuilder(NativeContract.ContractManagement.Id, Prefix_NextAvailableId);
+            StorageItem item = snapshot.TryGet(key);
+            return (int)(BigInteger)item;
+        }
+
+        private static void SetLastUsedContractId(DataCache snapshot, int newId)
+        {
+            StorageKey key = new KeyBuilder(NativeContract.ContractManagement.Id, Prefix_NextAvailableId);
+            StorageItem item = snapshot.GetAndChange(key);
+            item.Set(newId);
+        }
+        
+        private static int GetNextAvailableId(DataCache snapshot)
+        {
+            StorageKey key = new KeyBuilder(NativeContract.ContractManagement.Id, Prefix_NextAvailableId);
+            StorageItem item = snapshot.GetAndChange(key);
+            int value = (int)(BigInteger)item;
+            item.Add(1);
+            return value;
+        }
+
+        public static async Task<(ContractState contractState, (string key, string value)[] storagePairs)> ProcessDownloadParamsAsync(
+            string contractHash,
+            string rpcUri,
+            uint stateHeight,
+            bool isBatchCommand)
+        {
+            if (!UInt160.TryParse(contractHash, out var contractHash_))
+            {
+                throw new ArgumentException($"Invalid contract hash: \"{contractHash}\"");    
+            }
+
+            if (!TransactionExecutor.TryParseRpcUri(rpcUri, out var uri))
+            {
+                throw new ArgumentException($"Invalid RpcUri value \"{rpcUri}\"");
+            }
+
+            if (isBatchCommand && stateHeight == 0)
```

Comment:
We should check for stateHeight != 0 in batch command and remove isBatchCommand param